tests/countries/test_central_african_republic.py

```diff
@@ -0,0 +1,155 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from itertools import chain
+from unittest import TestCase
+
+from holidays.constants import WORKDAY
+from holidays.countries.central_african_republic import CentralAfricanRepublic, CF, CAF
+from tests.common import CommonCountryTests
+
+
+class TestCentralAfricanRepublic(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1959, 2050)
+        super().setUpClass(CentralAfricanRepublic, years=years, years_non_observed=years)
+        cls.workday_holidays = CentralAfricanRepublic(
+            years=years, categories=WORKDAY, islamic_show_estimated=False
+        )
+        cls.no_estimated_holidays = CentralAfricanRepublic(
+            years=years, islamic_show_estimated=False
+        )
+
+    def test_country_aliases(self):
+        self.assertAliases(CentralAfricanRepublic, CF, CAF)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(CentralAfricanRepublic(years=1958))
+
+    def test_new_years_day(self):
+        name = "Jour de l'an"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1959, 2050)))
```

Comment:
```suggestion
        self.assertHolidayName("Jour de l'an", (f"{year}-01-01" for year in range(1959, 2050)))
```
And same for similar holidays below.